src/uu/sort/src/sort.rs

```diff
@@ -696,9 +697,15 @@ impl KeyPosition {
             .ok_or_else(|| format!("invalid key {}", key.quote()))?;
         let char = field_and_char.next();
 
-        let field = field
-            .parse()
-            .map_err(|e| format!("failed to parse field index {}: {}", field.quote(), e))?;
+        let field = match field.parse::<usize>() {
+            Ok(field) => field,
+            Err(e) => {
+                match e.kind() {
+                    IntErrorKind::PosOverflow => usize::MAX,
+                    _ => return Err(format!("failed to parse field index {}", field.quote())),
+                }
+            },
+        };
```

Comment:
more rust idiomatic:

```suggestion
    let field = field.parse::<usize>()
    .unwrap_or_else(|e| match e.kind() {
        IntErrorKind::PosOverflow => usize::MAX,
        _ => panic!("failed to parse field index {}", field.quote()),
    });
```

Reply:
I like this better but is there a way to get away from `panic!` here? Assuming it would be better to unwind the error over panicing?

The contrib file also mentions we shouldn't panic.
https://github.com/uutils/coreutils/blob/main/CONTRIBUTING.md#dont-panic

Reply:
May I suggest this:
```rs
        let field = match field.parse::<usize>() {
            Ok(f) => f,
            Err(e) if *e.kind() == IntErrorKind::PosOverflow => usize::MAX,
            Err(_) => return Err(format!("failed to parse field index {}", field.quote()))
        };
```

Reply:
I think this looks the cleanest, returns the old behavior (assuming that is what we want) and gets rid of the multiple matches.

Reply:
Oh yeah, my bad. Sorry, no crash indeed

Reply:
```rust
let field = match field.parse::<usize>() {
    Ok(f) => f,
    Err(e) if *e.kind() == IntErrorKind::PosOverflow => usize::MAX,
    Err(e) => {
        return Err(format!(
            "failed to parse field index {} {}",
            field.quote(),
            e
        ))
    }
};
```

I've refactored to this. I believe this PR should be good to go now.